src/main/java/dev/openfeature/sdk/OpenFeatureAPI.java

```diff
@@ -82,7 +91,8 @@ public void setProvider(FeatureProvider provider) {
         if (provider == null) {
             throw new IllegalArgumentException("Provider cannot be null");
         }
-        defaultProvider = provider;
+        shutdownProvider(this.defaultProvider);
+        initializeProvider(provider);
```

Comment:
This is spec conformant as the spec is currently written, but I believe we may want to reword [1.1.2.2](https://openfeature.dev/specification/sections/flag-evaluation/#requirement-1122) slightly to emphasize that the "old" provider should not start until the "new" provider is done.

That would mean changing this slightly so that the shutdown of the old provider is a callback executed after initialization (or something similar). This way there is no "gap" in availability when providers are switched (though switching providers is perhaps in itself a bit of a niche use-case).

Reply:
I was thinking the same, but wanted to be close to the requirements. I will change this to reflect the intention and potential rewording

Reply:
see https://github.com/open-feature/spec/pull/193

Reply:
changed

Reply:
I'm sorry about all the back and forth on this @lopitz ! The last thing I want to do is waste your time... but it still seems like there's a bit of debate on this one. I'm starting to lean back to the previous implementation after the comments here ([particularly this one](https://github.com/open-feature/spec/pull/193#discussion_r1213344231)). I'd appreciate your opinion on this issue after working in this area.

Reply:
I've merged the spec PR above, which in the end contained mostly editorial changes. I think your original implementation (use the initialization provider immediately) is now more correct. Sorry about that again!

Reply:
don't worry. i think, writing the spec and forseeing all cases is not easy and if we gain knowledge in the implementation, we should address this in the spec - as we did. from my perspective, everything went the correct way and it wasn't annoying 😄